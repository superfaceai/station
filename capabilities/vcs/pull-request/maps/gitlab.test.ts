import { SuperfaceTest } from '@superfaceai/testing-lib';

describe(`vcs/pull-request/gitlab}`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest();
  });

  describe('PullRequest', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          profile: 'vcs/pull-request',
          provider: 'gitlab',
          useCase: 'PullRequest',
          input: {
            owner: 'Jakub-Vacek',
            repo: 'empty-test',
            identifier: 1,
          },
        })
      ).resolves.toMatchSnapshot();
    });

    it('should map error', async () => {
      await expect(
        superface.run({
          profile: 'vcs/pull-request',
          provider: 'gitlab',
          useCase: 'PullRequest',
          input: {
            owner: 'Jakub-Vacek',
            repo: 'empty-test',
            identifier: 9999,
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
