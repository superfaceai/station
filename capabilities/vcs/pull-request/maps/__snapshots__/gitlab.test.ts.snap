// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/pull-request/gitlab} PullRequest should map error 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
AST Path: definitions[0].statements[2].responseHandlers[0].statements[1]
Original Map Location: Line 20, column 7",
}
`;

exports[`vcs/pull-request/gitlab} PullRequest should perform successfully 1`] = `
Object {
  "value": Object {
    "id": 1,
    "sha": "8c64ce23d626c5bf345ce90fa5af329569d62c9a",
    "title": "Update README.md",
    "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/1",
  },
}
`;
